### DISCUSSION TITLE
知网辅助增强

### DISCUSSION THREAD

**Discussion by Kenshin** [Kenshin/simpread#3537] [插件说明]
描述
---

> 适配知网。

地址
---

[https://simpread.ksria.cn/plugins/details/yHqalbdeNq](https://simpread.ksria.cn/plugins/details/yHqalbdeNq)

使用视频
---

https://user-images.githubusercontent.com/81074/158159023-8aed1380-17ed-426f-a414-c35119d45ee2.mp4

说点不一样的
---

知网是我从开发简悦以来适配最困难的一个网站，不是说它有多难适配，而是我无法进入，所以辗转了很多方式，得到了很多同学们以及简悦用户的支持，才能完成这个插件。

感谢那些为此付出时间和心血的简悦用户，特意说下简悦用户 [@ilovepaper](https://github.com/Kenshin/simpread/discussions/2103#discussioncomment-2349356)，一共为此：18次回复与15封邮件，感谢付出 👏 👏

快捷键
---

> 支持全局快捷键 `c k`

使用前提
---

1. 知网的 URL 比较特殊，所以此方式没有使用站点适配规则，而是使用了插件来实现。

2. 随便进入一个知网页面，类似下面截图的方式进入的 HTML 页面

   ![image](https://user-images.githubusercontent.com/81074/158152098-6fce787b-794d-4494-9bf1-a826910f94d7.png)

3. 使用账户密码（非专线）登录的页面，可能需要等到图片全部加载完才能使用。

优化的模块
---

因知网的页面结构与其他网站不太一样，所以专门针对如下模块做了优化

1. 标题

    ![image](https://user-images.githubusercontent.com/81074/158155197-9210b23f-71ba-4d7b-aa18-acd066e05cee.png)

2. 作者标题

    ![image](https://user-images.githubusercontent.com/81074/158155231-3b8eae43-951b-490a-9f94-7d3dbc1fd822.png)

3. 参考来源

    ![image](https://user-images.githubusercontent.com/81074/158155136-e8d7a21b-acd7-4937-9b96-2dad6f0e8dff.png)

4. 注脚

    ![image](https://user-images.githubusercontent.com/81074/158155074-50dfa7dd-d3c3-42d6-9518-7bb4abacb4c7.png)

5. 目录（可点击）

    ![image](https://user-images.githubusercontent.com/81074/158154999-bfb5ae7a-c95e-4fde-b2e1-4a4a51326bed.png)

选项
---

> 随便进入一个知网的 HTML **阅读模式 → 右下角触发器 → 动作 → 插件触发器**

![image](https://user-images.githubusercontent.com/81074/158152134-556e4c40-2aee-4489-87da-4c55a8d04fd7.png)

功能说明
---

### 白名单

只有**输入的域名匹配当前页面**时才能自动优化知网页面，否则只能使用手动方式进入（支持快捷键）。

**主域名例子**：

1. `abc.com` 匹配 `xxxx.abc.com` `xxxx.yyy.abc.com`

2. `cnki.net` 匹配 `https://kns.cnki.net`

![image](https://user-images.githubusercontent.com/81074/158152704-b3064083-9158-4e4c-b408-fc60c6724802.png)

### 自动优化

当白名单中的内容与当前页面匹配时，开启此选项会自动优化知网页面。

![image](https://user-images.githubusercontent.com/81074/158153495-9733a8e5-5b2e-4bcc-ad1a-863ca549582d.png)

### 手动模式

在进入阅读模式后，可试试使用：

1. 右下角触发器 → 动作 → 插件触发器 → 知网辅助增强 开启优化

2. 支持快捷键 `c k`

更新日志
---

### 1.0.1

- `Fix` 标题获取失败的问题。

### 1.0.0

- 优化模块
    - 标题
    - 作者标题
    - 目录
    - 参考来源
    - 注脚

- 快捷键支持 `c k`

- 白名单
  
- 自动优化

版权声明
---

1. 简悦仅提供阅读界面优化且**仅能在知网支持 HTML 阅读功能以及支持 HTML阅读时使用**，请遵守知网以及版权方的相关规定。

   ![image](https://user-images.githubusercontent.com/81074/158789125-dcfece49-239a-44d1-82bf-fcfa7243b3af.png)

2. 当使用者有权限可以访问时才能使用此功能，简悦并不会提供任何途径给无权限者使用此功能的方式。

关联
---

简悦不仅为知网适配了，还包括很多其它科研类网站，细节 [请看这里](https://github.com/Kenshin/simpread/discussions/2103)。

引申
---

> 安装插件或带有插件触发器的插件失效的 [解决方案](https://github.com/Kenshin/simpread/discussions/2342)

**Comment by MartinYuan1995** [comment]


**Reply by Kenshin** [reply]
嗨，给出的信息太少我没法判断。

1. 其它页面是否可以标注？e.g. https://sspai.com/post/69972
2. 稍后读是否可以正常打开？
3. 建议截图或者录屏
4. 只是知网的某个页面？还是请给出出问题的标题截图

另外，当你看到这个信息，请新建一个 Issues 来描述你的问题 https://github.com/kenshin/simpread/issues/new
